pythonPackages.shtab: init at 1.4.2

michaeladler opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

[shtab](https://github.com/iterative/shtab) is a tool to easily generate bash and zsh completions for Python and even **non-Python** applications.
The tests are failing (and are therefore skipped) because the tests invoke bash and this version of bash does not come with completion features (compgen not found). shtab works fine though, I have already generated completions for non-trivial applications. 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

michaeladler committed:
pythonPackages.shtab: init at 1.4.2

SuperSandro2000 commented on pkgs/development/python-modules/shtab/default.nix:
```diff
@@ -0,0 +1,34 @@
+    sed -i -e 's/setuptools_scm\[toml\]>=[0-9\.]*//' setup.cfg
+  '';
+
+  doCheck = false;
```
Please add a pythonImportsCheck.

SuperSandro2000 commented on pkgs/development/python-modules/shtab/default.nix:
```diff
@@ -0,0 +1,34 @@
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "use_scm_version=True" "version='${version}'"
+    sed -i -e 's/setuptools>=[0-9\.]*;//' setup.cfg
+    sed -i -e 's/setuptools_scm\[toml\]>=[0-9\.]*//' setup.cfg
```
Do we really need those overrides? They are both >= so this should work with our never versions.

michaeladler replied:
It's not about the versions, it's about removing setuptools_scm as this requires having network connectivity during the build which fails in the sandbox.
But I don't need to remove setuptools, so I will keep that one in. I only need to remove the setuptools_scm string [here](https://github.com/iterative/shtab/blob/eb60f65335bc66ba36f5cf654f5f4fb4b9b8f2f8/setup.cfg#L69).

SuperSandro2000 replied:
Thats strange. setuptools_scm is used a lot in pythonPackages and I never saw that it requires internet connectivity. What does it try to download? Do we need to add that afterwards?

michaeladler replied:
Reading the output carefully I realized that setuptools_scm itself was missing! Adding it to the buildInputs fixed the problem. Thanks! I have updated the PR.

SuperSandro2000 commented:
> the tests invoke bash and this version of bash does not come with completion features (compgen not found).

Please try bashInteractive.

michaeladler commented:
> > the tests invoke bash and this version of bash does not come with completion features (compgen not found).
> 
> Please try bashInteractive.

Thanks, that was great advice and fixed the tests! I have updated the MR now.

michaeladler commented:
I guess this pull request is obsolete now since @fabaff added shtab in commit 6c3e84d2cb14610cf053299400442a30ac6f1ae5 

@fabaff I noticed you disabled the bash tests. You might want to take a look here, since I got all tests running thanks to @SuperSandro2000 good advice.
